connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -191,6 +193,17 @@ public boolean isSinkConnector() {
         return SinkConnector.class.isAssignableFrom(connector.getClass());
     }
 
+    public static ConnectorType connectorType(String clsname) {
+        if (null == clsname || clsname.isEmpty())
+            throw new IllegalArgumentException("Cannot determine the type for empty class");
+        try {
+            return SinkConnector.class.isAssignableFrom(Class.forName(clsname)) ?
+                ConnectorType.SINK : ConnectorType.SOURCE;
+        } catch (Exception e) {
+            throw new ConnectException("Cannot determine the type for " + clsname);
+        }
+    }
+
```

Comment:
Maybe you're in the middle of changes, but IIUC this logic is only called from within the `Worker.connectorType(...)` method. Since it may not be safe to be called in other places (due to the classloaders), how about moving the logic of checking the class/instance into `Worker.connectorType(...)` method and removing this method altogether?